---
title: "Pteropod Reference DB"
author: "Zack Gold"
date: "2024-09-10"
output: html_document
---
```{r}
library(tidyverse)
library(DECIPHER)
library(here)
```

# Leray CO1 rCRUX

```{r}
CO1_tax <- read.table("/Users/zgold/Downloads/UCLA_download_20240702/combined/rcRUX/Reference\ Databases/Big-databases_9-23/CO1_combined/blast_seeds_output/CO1_combined_taxonomy.txt", sep="\t", header=T)

CO1_fasta <- readDNAStringSet("/Users/zgold/Downloads/UCLA_download_20240702/combined/rcRUX/Reference\ Databases/Big-databases_9-23/CO1_combined/blast_seeds_output/CO1_combined.fasta")

```

```{r}
CO1_tax %>% 
  filter(., str_detect(taxonomic_path,"Pteropoda")) -> pteropods

pteropods$taxonomic_path %>%  unique()
```


```{r}
CO1_fasta[pteropods$accession] -> pteropod_fasta

pteropod_fasta
unique(pteropod_fasta) -> unique_pteropod_seqs

setdiff(names(pteropod_fasta),names(unique_pteropod_seqs)) -> non_unique_seqs

pteropod_fasta[non_unique_seqs] -> non_unique_fasta
  
View(non_unique_fasta)

as.data.frame(non_unique_fasta) %>% as_tibble(rownames = "accession") %>% 
  left_join(pteropods) -> non_unique_pteropod_df

names(non_unique_fasta) <- non_unique_pteropod_df$taxonomic_path

non_unique_pteropod_df %>%  
 ungroup() %>% 
  mutate(idx = match(x, unique(x))) %>% 
  select(-accession) %>% 
  distinct()
```

```{r}
BrowseSeqs(non_unique_fasta,highlight=0)

d <- DistanceMatrix(non_unique_fasta, correction="none", verbose=FALSE)
c <- TreeLine(myDistMatrix=d, method="UPGMA", cutoff=.01, showPlot=TRUE, verbose=FALSE)

clusters <- Clusterize(non_unique_fasta, cutoff=0.001, invertCenters=FALSE)

apply(clusters, 2, function(x) max(abs(x))) # number of clusters

clusters %>% arrange(cluster)


```

```{r}
clusters %>% 
  add_rownames(., var = "Name") %>% 
  arrange(cluster) %>% 
  dplyr::select(cluster,Name) %>% 
  distinct() -> cluster_IDS

 cluster_IDS %>% 
   filter(., !is.na(Name)) %>% 
  group_by(cluster) %>% summarise(n=sum(n())) %>% filter(n>1) -> duplicate_reference_cluster

cluster_IDS %>% 
   filter(., !is.na(Name)) %>% 
  filter(., !cluster %in% duplicate_reference_cluster$cluster) %>% 
  dplyr::select(cluster, Species = Name)-> trust_worthy_IDS

cluster_IDS %>% 
  filter(., cluster %in% duplicate_reference_cluster$cluster)-> sequence_clusters_multiple_names

# Creseis can not be id'd to species
```

# Kelly 16S rCRUX

```{r}
K16S_tax <- read.table("/Users/zgold/Downloads/UCLA_download_20240702/combined/rcRUX/Reference\ Databases/Databases_4_18_corrected_9-25/Kelly_16S/blast_seeds_output/Kelly_16S_taxonomy.txt", sep="\t", header=F) %>% select(accession=V1, taxonomic_path=V2)

K16S_fasta <- readDNAStringSet("/Users/zgold/Downloads/UCLA_download_20240702/combined/rcRUX/Reference\ Databases/Databases_4_18_corrected_9-25/Kelly_16S/blast_seeds_output/Kelly_16S.fasta")

```

```{r}
K16S_tax %>% 
  filter(., str_detect(taxonomic_path,"Pteropoda")) -> pteropods_16S

pteropods_16S$taxonomic_path %>%  unique()
```

# MiDeca rCRUX

```{r}
MiDeca_tax <- read.table("/Users/zgold/Downloads/UCLA_download_20240702/combined/rcRUX/Reference\ Databases/Databases_4_18_corrected_9-25/MiDeca/blast_seeds_output/MiDeca_taxonomy.txt", sep="\t", header=F) %>% select(accession=V1, taxonomic_path=V2)

MiDeca_fasta <- readDNAStringSet("/Users/zgold/Downloads/UCLA_download_20240702/combined/rcRUX/Reference\ Databases/Databases_4_18_corrected_9-25/MiDeca/blast_seeds_output/MiDeca.fasta")

```

```{r}
MiDeca_tax %>% 
  filter(., str_detect(taxonomic_path,"Euphausiacea")) -> MiDeca_16S

MiDeca_16S$taxonomic_path %>%  unique()
```

